sourcetracker/_plot.py

```diff
@@ -10,16 +10,306 @@
+                g[i] = sns.heatmap(prop.iloc[:, i:i + 1],
+                                   vmin=0, cmap=self.color,
+                                   cbar=False, annot=annot,
+                                   ax=axes[i])
+                g[i].set_xlabel("")
+                g[i].set_ylabel("")
+                g[i].set_yticks([])
+        for ax in g:
+            tl = ax.get_xticklabels()
+            ax.set_xticklabels(tl, rotation=0)
+            tly = ax.get_yticklabels()
+            ax.set_yticklabels(tly, rotation=0)
+        if normalized:
+            if unknowns:
+                add_line = tra + "_pairedheatmap_normalized.png"
+                plt.savefig(os.path.join(self.file,
+                                         self.out_name + add_line))
+            else:
+                add_line = tra + "_pairedheatmap_nounknown_normalized.png"
+                plt.savefig(os.path.join(self.file,
+                                         self.out_name + add_line))
+        else:
+            if unknowns:
+                add_line = tra + "_pairedheatmap.png"
+                plt.savefig(os.path.join(self.file,
+                                         self.out_name + add_line))
+            else:
+                add_line = tra + "_pairedheatmap_nounknowns.png"
+                plt.savefig(os.path.join(self.file,
+                                         self.out_name + add_line))
+
+    def ST_Stacked_bar(self, unknowns=True, x_lab="Sink",
+                       y_lab="Source Proportion", coloring=[], flipped=False):
+        """
+        Creates a Stacked bar plot for the user with direct png save function
+        Parameters
+        ----------
+        unknowns bool
+            removes unknown column
+        xlabel string
+            x label
+        ylabel string
+            y label
+        coloring string list
+            string list of colors to encode the bar plot.
+            must have and equal number of colors as sources
+        flipped bool
+            flips x and y axis
+        returns
+        --------
+        none
+        outputs a stacked bar visualization in a PNG
+
+        color example list
+
+        '#1f77b4'Blue, '#ff7f0e'Orange, '#2ca02c'Green, '#d62728'Red,
+        '#9467bd'Purple, '#8c564b'Brown, '#e377c2'Pink, '#7f7f7f'Grey,
+        '#bcbd22'Gold, '#17becf'Cyan
+        make sure to use contrasting colors in order better illuminate
+        your data above are some example codes to use
+        """
+        prop = self.mpm
+        if not unknowns:
+            prop = prop.drop(['Unknown'], axis=1)
+            prop = prop.div(prop.sum(axis=1), axis=0)
+        if flipped:
+            prop = prop.T
+            y_lab_flip = x_lab
+            x_lab_flip = y_lab
+            y_lab = y_lab_flip
+            x_lab = x_lab_flip
+            prop = prop.div(prop.sum(axis=1), axis=0)
+        prop = prop.reset_index()
+        if len(coloring) != (prop.shape[1]-1):
+            coloring = []
```

Comment:
This is currently a silent error. I think we should raise a ValueError if len(coloring) != (prop.shape[1]-1).

Reply:
Honestly I just avoided the problem entirely by reverting to default color scheme. Where would I look in ST for an example of where something like this would be raised. I just don't want to boot the user from their run because they could still make the plot with the data anyway over the number of colors they chose.

Reply:
Reverting back to the default color scheme is what makes this a silent error. 

Imagine the User that really wants the non-default color scheme but puts the wrong number of colors. The user would run their code and then when they got their result it would have the default color scheme. They wouldn't know what was going on, They passed in a list of colors and from their prospective the code ran successfully but completely ignored their color selections.

Reply:
![ben kenobi](https://github.com/caporaso-lab/sourcetracker2/assets/128522580/6522af88-cedf-4f85-bed3-3037d8a7da73)
I have run into this issue before but when I have been running this kind of code before I would rather have the code give me the result anyway because I can change the number later and dont want to get in the way of results. What if there was a way to tell the user this? Someway to tell them how many colors they have and how many they need?

Reply:
My take on errors is that they give us a way to explain very clearly to the user what went wrong. 

To me I would rather have to re-run a command and understand what went wrong, than get a result that for a unknown reason isn't "listening" to my commands. 

> because I can change the number later and dont want to get in the way of results.

I think this is over assuming what the user will take away from the result. I believe that the average user will run this code, get the default colors and assume the code is broken. From their prospective the code didn't give an error but the code also "didn't work" because it didn't color the plot they way they were expecting.  I don't know that the majority of users would assume that they did anything wrong. 

> What if there was a way to tell the user this? Someway to tell them how many colors they have and how many they need?

You could give a warning and not fail out but qiime2 ecosystem tends to avoid that because 1) cli vs python errors are hard to format. 2) users do not typically read warnings very much.